### PR TITLE
feat: introduce `TableNameCache` & `TableInfoCache` & `TableRouteCache`

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

1. Introduce `TableNameCache`
2. Introduce `TableInfoCache`
3. Introduce `TableRouteCache`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LVGTM

File: src/common/meta/src/cache/container.rs [hunk 1]
```diff
@@ -126,6 +126,15 @@ where
         Ok(())
     }
 
+    /// Returns true if the cache contains a value for the key.
+    pub fn contains_key<Q>(&self, key: &Q) -> bool
```
Comment (zhongzc) [inline src/common/meta/src/cache/container.rs:130]: A great function to make unit tests less weird!

File: src/common/meta/src/cache/table/table_name.rs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+use crate::cache::{CacheContainer, Initializer};
+use crate::error;
+use crate::error::Result;
+use crate::instruction::CacheIdent;
+use crate::key::table_name::{TableNameKey, TableNameManager, TableNameManagerRef};
+use crate::kv_backend::KvBackendRef;
+use crate::table_name::TableName;
+
+/// [TableNameCache] caches the [TableName] to [TableId] mapping.
+pub type TableNameCache = CacheContainer<TableName, u32, CacheIdent>;
```
Comment (zhongzc) [inline src/common/meta/src/cache/table/table_name.rs:31]: Can we use the alias for `TableId` here?

**Review by zhongzc — APPROVED** [review]

File: src/common/meta/src/cache/table/table_name.rs (hunk 2 above)
Comment (WenyXu) [reply src/common/meta/src/cache/table/table_name.rs:31]: Good point, it should be `TableId`

**Pull request merged**